external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/src/camessagehandler.c

```diff
@@ -872,7 +912,23 @@ static CAResult_t CAReceivedPacketCallback(const CASecureEndpoint_t *sep,
     else
 #endif
     {
-        CAQueueingThreadAddData(&g_receiveThread, cadata, sizeof(CAData_t));
+        // NOTE: This is test logic for CONPRO-1172
+        // Enqueue request only if queue size is under threshold size.
+        // This logic needs refactoring considering the fact that this queue
+        // is used for responses as well so accounting total queue size for
+        // request drop is not optimal.
+        size_t queueSize = CAQueueingThreadGetQueueSize(&g_receiveThread);
+        if (cadata->requestInfo == NULL || queueSize <= RECEIVE_QUEUE_MAX_SIZE)
```

Comment:
In case that g_receiveThread is NULL, the following if statement is true.
I think you need another checkpoint logic to see whether g_receiveThread is null or not.